internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -0,0 +1,285 @@
+				"associated_domains":           federatedSettingsIdentityProvider[i].AssociatedDomains,
+				"associated_orgs":              FlattenAssociatedOrgs(federatedSettingsIdentityProvider[i].AssociatedOrgs),
+				"audience_uri":                 federatedSettingsIdentityProvider[i].AudienceUri,
+				"display_name":                 federatedSettingsIdentityProvider[i].DisplayName,
+				"issuer_uri":                   federatedSettingsIdentityProvider[i].IssuerUri,
+				"okta_idp_id":                  federatedSettingsIdentityProvider[i].OktaIdpId,
+				"pem_file_info":                FlattenPemFileInfo(*federatedSettingsIdentityProvider[i].PemFileInfo),
+				"request_binding":              federatedSettingsIdentityProvider[i].RequestBinding,
+				"response_signature_algorithm": federatedSettingsIdentityProvider[i].ResponseSignatureAlgorithm,
+				"sso_debug_enabled":            federatedSettingsIdentityProvider[i].SsoDebugEnabled,
+				"sso_url":                      federatedSettingsIdentityProvider[i].SsoUrl,
+				"status":                       federatedSettingsIdentityProvider[i].Status,
+			}
+		}
+	}
+
+	return federatedSettingsIdentityProviderMap
+}
+
+func FlattenAssociatedOrgs(associatedOrgs []admin.ConnectedOrgConfig) []map[string]any {
+	var associatedOrgsMap []map[string]any
+
+	if len(associatedOrgs) == 0 {
+		return nil
+	}
+	associatedOrgsMap = make([]map[string]any, len(associatedOrgs))
+
+	for i := range associatedOrgs {
+		if associatedOrgs[i].UserConflicts == nil {
+			associatedOrgsMap[i] = map[string]any{
+				"domain_allow_list":          associatedOrgs[i].DomainAllowList,
+				"domain_restriction_enabled": associatedOrgs[i].DomainRestrictionEnabled,
+				"identity_provider_id":       associatedOrgs[i].IdentityProviderId,
+				"org_id":                     associatedOrgs[i].OrgId,
+				"post_auth_role_grants":      associatedOrgs[i].PostAuthRoleGrants,
+				"role_mappings":              FlattenAuthFederationRoleMapping(associatedOrgs[i].RoleMappings),
+				"user_conflicts":             nil,
+			}
+		} else {
+			associatedOrgsMap[i] = map[string]any{
+				"domain_allow_list":          associatedOrgs[i].DomainAllowList,
+				"domain_restriction_enabled": associatedOrgs[i].DomainRestrictionEnabled,
+				"identity_provider_id":       associatedOrgs[i].IdentityProviderId,
+				"org_id":                     associatedOrgs[i].OrgId,
+				"post_auth_role_grants":      associatedOrgs[i].PostAuthRoleGrants,
+				"role_mappings":              FlattenAuthFederationRoleMapping(associatedOrgs[i].RoleMappings),
+				"user_conflicts":             FlattenFederatedUser(associatedOrgs[i].UserConflicts),
+			}
+		}
+	}
+
+	return associatedOrgsMap
+}
+
+type mRoleAssignmentV2 []admin.RoleAssignment
+
+func (ra mRoleAssignmentV2) Len() int      { return len(ra) }
+func (ra mRoleAssignmentV2) Swap(i, j int) { ra[i], ra[j] = ra[j], ra[i] }
+func (ra mRoleAssignmentV2) Less(i, j int) bool {
+	compareVal := strings.Compare(*ra[i].OrgId, *ra[j].OrgId)
```

Comment:
```suggestion
	compareVal := strings.Compare(ra[i].GetOrgId(), ra[j].GetOrgId())
```
Better to avoid directly accessing pointer values. 2 similar cases below in this function.

Reply:
Thanks, changed all similar cases